src/nvim/terminal.c

```diff
@@ -405,6 +405,7 @@ void terminal_enter(void)
   // erase the unfocused cursor
   invalidate_terminal(s->term, s->term->cursor.row, s->term->cursor.row + 1);
   showmode();
+  win_redr_status(curwin);
```

Comment:
This should probably have a small comment explaining why this work-around was introduced. After all you said yourself that it *should* not be needed to redraw the status line when switching to terminal mode. :wink:

Reply:
Sure ^^. I wasn't quite sure if this is actually the right way to do this so I just wanted to put it up for debate. Let me amend it real quick :)

Reply:
The bug was honestly driving me nuts in the past few days. "Y U NO ENTER TERMINAL MODE?… oh you did…" I heavily depend in some visual feedback and with a high enough resolution `showmode` sometimes is way too far away from where the action is happening. For me this was a good enough reason to search for a workaround ^^. Hopefully someone has a better idea :)

Reply:
Status redrawing is quite a bit weird (when debugging #6619, I've seen the statusbar being redrawn without seeing the relevant functions being scheduled or even entered by the debugger), so it is one place where a little _ad-hoc_ logic doesn't hurt. Still, having a comment is good :)

Reply:
Agreed. I'm currently test driving the patch in my setup. Doesn't seem to cause any issues so far.